### PR TITLE
Use POST endpoint instead of expand the cluster list

### PR DESCRIPTION
**Pull request by joselsegura**
# Description

`org_overview` endpoint currently used the GET endpoint to retrieve all the cluster reports from Aggregator. It can lead in some situations to very long URLs that can be handled differently depending on configuration.

This change makes the endpoint to use the POST endpoint version to retrieve the reports for a list of clusters, which will avoid the possible URL problems

## Type of change

- Refactor (refactoring code, removing useless files)

## Testing steps

Regular CI

## Checklist
* [x] `make before_commit` passes
* [x] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [x] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by dpensi — APPROVED** [review]

**Review by juandspy — APPROVED** [review]

File: server/server.go [hunk 1]
```diff
@@ -407,6 +408,53 @@ func (server HTTPServer) readAggregatorReportForClusterList(
...
+		handleServerError(writer, err)
+		return nil, false
+	}
+	// #nosec G107
+	aggregatorResp, err := http.Post(aggregatorURL, "application/json", bytes.NewBuffer(body))
+	if err != nil {
+		handleServerError(writer, err)
+		return nil, false
+	}
+
```
Comment (tisnik) [inline server/server.go:431]: probably will be better to split the function at this line. The reason - free the `body` + associated `Buffer` as fast as possible as it is memory intensive.

**Review by tisnik — COMMENTED** [review]
One comment, LGTM otherwise.

File: server/server.go (hunk 1 above)
Comment (joselsegura) [reply server/server.go:431]: Makes sense. Updating

**Review by tisnik — APPROVED** [review]
LGTM

**Pull request merged**